src/azul/service/manifest_service.py

```diff
@@ -703,10 +705,62 @@ def write_to(self, output: IO[str]) -> Optional[str]:
             url = furl(config.service_endpoint(),
                        path=f'/repository/files/{uuid}',
                        args=dict(version=version, catalog=self.catalog))
+            name = self._process_path(name)
             output.write(f'url={self._option(url.url)}\n'
                          f'output={self._option(name)}\n\n')
         return None
 
+    @classmethod
+    def _process_path(cls, path: str) -> str:
+        """
+        >>> f = CurlManifestGenerator._process_path
+        >>> f('foo/bar/\\x1F/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/\\x1f/file', \\x00 to \\x1F characters and `\\` are invalid
+
+        >>> f('foo/bar/\\/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/\\\\/file', \\x00 to \\x1F characters and `\\` are invalid
+
+        >>> f('foo/bar/COM6/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The use of {'COM6'} is disallowed in file name 'foo/bar/COM6/file'
+
+        >>> f('foo/bar/ / baz/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid name sequence in file name 'foo/bar/ / baz/file'
+
+        >>> f('foo/bar?/fi*le|')
+        'foo/bar_/fi_le_'
+
+        >>> f('foo/bar/file.fastqgz')
+        'foo/bar/file.fastqgz'
+        """
+        for character in path:
+            reject(character == '\\' or character < chr(32),
+                   f'Invalid file name {path!r}, \\x00 to \\x1F characters and `\\` are invalid')
+        path = re.sub(r'[<>:"|?*]', '_', path)
+        valid_sequence = re.compile(r'''
+            (\.?[^./ ])([^/]*[^./ ])? # match paths not followed or preceeded by '/', ' ', '.' zero or more times
+            (/(\.?[^./ ])([^/]*[^./ ])?)* # except for '/', ' ', '.' being in between other sequences not at boundaries
+        ''', re.X)
+        path_re = valid_sequence.fullmatch(path)
+        reject(path_re is None,
+               f'Invalid name sequence in file name {path!r}')
+        names = {
+            'CON', 'PRN', 'AUX', 'NUL',
+            *(f'{cmd}{i}' for cmd in ['COM', 'LPT'] for i in range(1, 10))
+        }
+        components = {*path.split('/')}
+        set_ = names.isdisjoint(components) or names.intersection(components)
+        require(set_ is True,
+                f'The use of {set_!r} is disallowed in file name {path!r}')
```

Comment:
Also, overloading the variable with `or` is hard to follow. So

```suggestion
        invalid_components = components.intersection(self.disallowed_path_components)
        require(invalid_components == {},
                f'Invalid path component', invalid_components)
```

Reply:
Note how tha eliminates the string interpolation by passing more args to `require`. Use that across the board.